### PR TITLE
Improve documentation organization, accuracy, and readability

### PR DESCRIPTION
**Pull request by wangyinz**
## Summary

- reorganize the documentation around clearer desktop, cluster, data, processing, and reference paths
- review and update user-manual and API pages while preserving important existing material
- simplify the README and onboarding guides so the short Docker happy path remains prominent
- correct the Spark Compose database service reference while documenting that `mspass-dbmanager` remains valid for the sharded configuration
- fix generated API-reference markup, including the ambiguous `elog` cross-reference

## Why

The documentation had outdated commands, broken or incomplete API coverage, inconsistent navigation, and some deployment guidance that no longer matched the shipped configuration. A preservation/readability pass also found that several initial rewrites made first-run instructions unnecessarily intimidating, so those pages were simplified and the original approachable commands and list structure restored.

## User impact

Users get clearer navigation, current deployment and data-management guidance, more complete API references, and shorter onboarding flows. Existing conceptual explanations, diagrams, examples, and important caveats are retained.

## Validation

- full Sphinx HTML build with the project `mspass` Conda environment
- all documentation pages rendered successfully
- ambiguous generated `elog` cross-reference warning eliminated
- only two expected offline intersphinx inventory warnings remained
- shipped Dask and Spark Compose files validated with `docker compose config --quiet`
- `git diff --check`

Local HTML output used for final review: `/tmp/mspass-docs-html-final-20260711/index.html`.

### CODE REVIEW

**Comment by wangyinz** [comment]
I completed a preservation-first corrective review after identifying that the earlier readability pass removed too much beginner-oriented material.

Corrective commits:

- `5630a2e2` restores detailed user guidance in the high-churn manual and Docker/Compose pages.
- `c5623686` restores onboarding explanations, the original quick-start teaching flow, and historical authorial content.

The review covered every changed documentation file. Material was restored unless it was demonstrably obsolete, incorrect, an unfinished placeholder, or replaced by a code-verified correction. Examples include restoring the full 913-line parallel-I/O chapter, the beginner error-handling philosophy, CRUD explanations, Docker port/mount/Jupyter guidance, virtual-cluster concepts and diagrams, GUI configuration explanations, and the original quick-start startup-output walkthrough.

Final validation: a clean forced Sphinx HTML build rendered every page successfully. The only warnings were the two offline external intersphinx inventories; the ambiguous generated `elog` warning remains fixed.

Review build: `/tmp/mspass-docs-html-preservation-final/index.html`.

File: docs/source/user_manual/data_object_design_concepts.rst [hunk 1]
```diff
@@ -137,41 +133,44 @@ Overview - Inheritance Relationships
...
-  but not always have python bindings.
+  but not always have Python bindings.
 
 | The remainder of this section discusses the individual components in
   the class hierarchy.
 
 BasicTimeSeries - Base class of common data characteristics
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-This base class object can be best viewed as an answer to this
```
Comment (pavlis) [inline docs/source/user_manual/data_object_design_concepts.rst:169]: The change has an incorrect usage with "says saying all ..."

File: docs/source/user_manual/FAQ.rst
Comment (pavlis) [inline docs/source/user_manual/FAQ.rst:1]: The AI added a lot of useful FAQ entries here.  We definitely need to accept them.

File: docs/source/user_manual/importing_tabular_data.rst [hunk 2]
```diff
@@ -20,136 +20,145 @@ means "table".
...
 - A special form of relational database in the seismology community is
   what is most properly called a "Datascope database" but which is commonly
-  referred to as a "Antelope database".   Technically "Datascope" is a
-  component of the larger "Antelope" software suite.   Those tables are important
+  referred to as an "Antelope database".  Technically, Datascope is a
+  component of the larger Antelope software suite.  Those tables are important
   as a number of regional seismic networks use Antelope as their database
-  engine.  Furthermore, the Array Network Facility of the Earthscope project
-  used Antelope and the database tables are still accessible at
-  the AUG web site `here <https://anf.ucsd.edu/tools/events/>`__.
```
Comment (pavlis) [inline docs/source/user_manual/importing_tabular_data.rst:38]: There is a permanant link with a doi to these data.   I need to look it up, but we whoudl replace the now dead link here that is retained with these changes to the doi link.

File: docs/source/user_manual/memory_management.rst
Comment (pavlis) [inline docs/source/user_manual/memory_management.rst:1]: This section needs a major rewrite as it contains some errors the AI has not fixed.   I'm not sure I agree with all the wording changes either, but since this needs a rewrite anyway I suggest we just take all these changes as they clean up a lot of other issues.

File: docs/source/user_manual/normalization.rst
Comment (pavlis) [inline docs/source/user_manual/normalization.rst:1]: The changes here seem to mostly be to many lines of code examples in this section.  It is very hard to know if the code changes are correct but the originals likely are wrong too.   I think we should just accept the changes in this file.

File: docs/source/user_manual/numpy_scipy_interface.rst
Comment (pavlis) [inline docs/source/user_manual/numpy_scipy_interface.rst:1]: The AI fixed a lot of things here.   Not sure I agree with some of the wording changes, but it is too difficult to fix them at this stage.  Like some other files it is hard to know if the suggesting changes to code examples are valid.  Ones I examined seem to be ok and correct deficiencies in the current version it would replace.

File: docs/source/user_manual/obspy_interface.rst [hunk 3]
```diff
@@ -124,7 +158,7 @@ Consider this small code fragment to apply a bandpass filter to an ObsPy :code:`
 .. code-block:: python
 
    from obspy import read
-   d = read('mydatafile')
+   d = read("mydatafile")[0]
```
Comment (pavlis) [inline docs/source/user_manual/obspy_interface.rst]:
```suggestion
   d = read("mydatafile")
```
That is essentially rejecting the change.   Not sure why the AI put that [0] there

File: docs/source/user_manual/research_computing_essay.rst
Comment (pavlis) [inline docs/source/user_manual/research_computing_essay.rst:1]: I think I managed to actually remove this file.  It should not have been in the repository.   Let's make sure I did that right.

File: docs/source/user_manual/signal_to_noise.rst [hunk 4]
```diff
@@ -75,30 +77,32 @@ window is called the :math:`L_\infty` norm defined as:
 
 .. math::
 
-  | \bf{x} |_{\infty} = max ( \mid x_{n_s}\mid ,  \mid x_{n_s + 1}\mid , \cdots , \mid x_{n_e}\mid )
+  \lVert \mathbf{x} \rVert_{\infty}
+  = \max ( \lvert x_{n_s}\rvert ,  \lvert x_{n_s + 1}\rvert , \ldots , \lvert x_{n_e}\rvert )
```
Comment (pavlis) [inline docs/source/user_manual/signal_to_noise.rst:81]: I have not easy way to verify if this latex code change is correct.   Hopefully it is.

**Review by pavlis — COMMENTED** [review]
I have some minor things I think should be resolved before we merge this, but the big picture is this fixes a lot more problems and mistakes than it creates.  i.e. there are likely mistakes I missed, but it nothing compared to what it fixes.   

@wangyinz look at the comments I left and consider implementing some of the minor changes I suggested.  One thing I think I did but can't tell for sure is remove a file called "research_computing_essay.rst".   That was something I started and decided to not include.  It must have gotten pushed to the repository in error.  Make sure it is removed with this pull request.

File: docs/source/user_manual/data_object_design_concepts.rst (hunk 1 above)
Comment (wangyinz) [reply docs/source/user_manual/data_object_design_concepts.rst:169]: Fixed in d8caa6f6: the duplicated “says saying” wording is removed while retaining the original point.

File: docs/source/user_manual/FAQ.rst
Comment (wangyinz) [reply docs/source/user_manual/FAQ.rst:1]: Thanks—retained the useful FAQ additions and their question-and-answer structure.

File: docs/source/user_manual/importing_tabular_data.rst (hunk 2 above)
Comment (wangyinz) [reply docs/source/user_manual/importing_tabular_data.rst:38]: Fixed in d8caa6f6. The dead ANF event link now points to the final archive DOI, https://doi.org/10.17611/DP/EB.1. I also found and corrected the same stale link in `continuous_data.rst`.

File: docs/source/user_manual/memory_management.rst
Comment (wangyinz) [reply docs/source/user_manual/memory_management.rst:1]: Addressed in d8caa6f6 after a fresh technical review. The page no longer promises that containers prevent host crashes or that Dask registers C++ object sizes; it now distinguishes managed memory from RSS, treats spill storage as finite, separates Dask thresholds from Spark behavior, accounts for workers and threads correctly, labels copy-count estimates as heuristics, notes key-skew/shuffle effects, and corrects the guidance on ordered data and dismantling ensembles.

File: docs/source/user_manual/normalization.rst
Comment (wangyinz) [reply docs/source/user_manual/normalization.rst:1]: Thanks—retained the normalization changes as reviewed.

File: docs/source/user_manual/numpy_scipy_interface.rst
Comment (wangyinz) [reply docs/source/user_manual/numpy_scipy_interface.rst:1]: Thanks—retained the NumPy/SciPy interface corrections as reviewed.

File: docs/source/user_manual/obspy_interface.rst (hunk 3 above)
Comment (wangyinz) [reply docs/source/user_manual/obspy_interface.rst]: Correction: you were right. I initially focused on the `Trace2TimeSeries` contract, but that converter is not what this example exercises. `obspy.read()` returns a `Stream`, native `Stream.filter()` processes every contained `Trace`, and the MsPASS `signals.filter` wrapper supports ensemble inputs through its Stream conversion decorators. Commit `29d18d7c` removes `[0]`, uses `stream.filter(...)`, and explains that indexing is needed only when the user intentionally wants one channel. For completeness, direct conversion still uses `Trace2TimeSeries` for one `Trace` and `Stream2TimeSeriesEnsemble` for a `Stream`.

File: docs/source/user_manual/research_computing_essay.rst
Comment (wangyinz) [reply docs/source/user_manual/research_computing_essay.rst:1]: Confirmed in d8caa6f6: `research_computing_essay.rst` is removed, and its obsolete exclusion was removed from `conf.py`. The branch also preserves the separate removal of the unfinished historical appendix essay.

File: docs/source/user_manual/signal_to_noise.rst (hunk 4 above)
Comment (wangyinz) [reply docs/source/user_manual/signal_to_noise.rst:81]: The RMS and peak equations are correct: the RMS denominator uses the inclusive sample count `n_e - n_s + 1`, and the peak definition matches the maximum absolute amplitude. The rendered LaTeX passes the Sphinx build. A deeper implementation audit did uncover a separate `PercAmplitude` defect: the `CoreSeismogram` overload does not normalize percentage inputs the way the `CoreTimeSeries` overload does, and both overloads have edge cases at 100%/empty input. That code issue is separate from the equation and should be corrected with focused C++ tests rather than folded into this documentation PR.

**Comment by wangyinz** [comment]
Addressed the human review feedback in `d8caa6f6`, the ObsPy follow-up in `29d18d7c`, and the Copilot review in `9b52e18d`. All eleven inline review threads now have responses and are resolved.

- Fixed the duplicated wording and replaced the dead ANF link with the final archive DOI in both affected pages.
- Re-reviewed `memory_management.rst` and corrected the concrete Dask/Spark, RSS, spill, worker/thread, heuristic, skew, and ensemble guidance issues.
- Corrected the ObsPy example to retain the `Stream`: native `Stream.filter()` processes every contained `Trace`, and the MsPASS `signals.filter` wrapper supports ensemble inputs through Stream conversion decorators.
- Corrected the matcher input contract to accept `Metadata`, added an unambiguous Sphinx type link, and aligned the documented list return contract with implementation.
- Corrected `rota` to `rtoa` in both locations and repaired the nearby ensemble-processing example so it processes each member and uses the current APIs.
- Confirmed removal of the unfinished essays and verified the RMS and peak equations and their LaTeX.
- Audited for similar documentation problems and fixed additional grammar, duplicate words, stale links, API names, type contracts, and code-example errors.

Validation: Python compilation and a runtime plain-`Metadata` matcher check pass; `git diff --check` passes; and a clean forced Sphinx HTML build succeeds. The only two warnings are unreachable external ObsPy and PyMongo intersphinx inventories.

File: docs/source/user_manual/obspy_interface.rst (hunk 3 above)
Comment (wangyinz) [reply docs/source/user_manual/obspy_interface.rst]: Turns out Codex was wrong about this, but it is now fixed in 29d18d7

File: python/mspasspy/db/normalize.py [hunk 5]
```diff
@@ -670,13 +670,13 @@ def find(self, mspass_object) -> tuple:
...
-          or SeismogramEnsemble.   If it is anything else (e.g. None)
-          this method will return a tuple [None, elog] with elog being
-          a PyErrorLogger with a posted message.
+        :param mspass_object: data object to match against the cached data.
+          It must be a valid MsPASS data object: TimeSeries, Seismogram,
+          TimeSeriesEnsemble, or SeismogramEnsemble.  For any other input
+          type, the method returns ``[None, elog]`` with a
+          :class:`~mspasspy.util.error_logger.PyErrorLogger` explaining the
+          failure.
+        :type mspass_object: Metadata
```
[the remark this answers is not in the document]
Comment (wangyinz) [reply python/mspasspy/db/normalize.py]: Fixed in `9b52e18d`. `DictionaryCacheMatcher.find` and `find_one` now state that they accept a `Metadata` instance, explicitly include atomic and ensemble subclasses, note that concrete matchers may require keys or narrower subtypes, and use the fully qualified Sphinx class reference. The similar-issue audit also aligned the related `EqualityMatcher` contract and corrected matcher return annotations/documentation from tuple to the two-element lists the implementations actually return. Python compilation, a runtime plain-`Metadata` matcher check, and the Sphinx build all pass.

File: python/mspasspy/seismic/gather.py [hunk 6]
```diff
@@ -152,33 +152,34 @@ def extractMemberMetadataFromOldEnsemble(ensemble_object):
...
+    that would never work if only the base class were constructed
+    (not allowed because of the virtual methods).  In particular, the
+    way the base class handles metadata is potentially confusing because
     the base class constructor only defines stubs for the data used to
     hold member and ensemble metadata
 
-    A design issue for discussionis is whether not a gather should implement
+    A design issue is whether a gather should implement
     the ator and rota functionality of BasicTimeSeries.   The cost is
     tiny up front and retrofitting after that fact might cause a lot of
```
[the remark this answers is not in the document]
Comment (wangyinz) [reply python/mspasspy/seismic/gather.py:184]: Fixed in `9b52e18d`: `rota` is now `rtoa`. The same typo also existed in the parallel-processing example and is corrected there. While checking that example, I fixed its invalid `db.name` parameter, changed the loop from repeatedly processing `member[0]` to `member[i]`, moved client setup out of the function body, and updated the nearby distributed-read call to the current API. Python compilation and the full Sphinx build pass.

File: python/mspasspy/db/normalize.py [hunk 7]
```diff
@@ -367,7 +367,7 @@ def find(self, mspass_object):
         """
         if not isinstance(mspass_object, Metadata):
             elog = PyErrorLogger()
-            message = "received invalid data.  Arg0 must be a valid MsPASS data object"
+            message = "received invalid data.  Arg0 must be a Metadata instance"
             elog.log_error(message, ErrorSeverity.Invalid)
```
[the remark this answers is not in the document]
Comment (wangyinz) [reply python/mspasspy/db/normalize.py:369]: Fixed in a9763ac4. DatabaseMatcher.find now returns [None, elog] immediately for non-Metadata input, with regression coverage for find and find_one. I also audited analogous matcher control flow and fixed the related MiniseedDBMatcher error-logger return, ArrivalMatcher filtering and cache defects, and OriginTimeMatcher candidate and optional-field defects, each with focused regressions.
